src/WPGraphQL.php

```diff
@@ -422,50 +424,115 @@ public function init_admin() {
+	}
 
+	/**
+	 * Sets up the default taxonomies to show_in_graphql.
+	 *
+	 * @param array  $args      Array of arguments for registering a taxonomy.
+	 * @param string $taxonomy  Taxonomy key.
+	 *
+	 * @return array
+	 */
+	public static function setup_default_taxonomies( $args, $taxonomy ) {
 		// Adds GraphQL support for categories.
-		if ( isset( $wp_taxonomies['category'] ) ) {
-			$wp_taxonomies['category']->show_in_graphql     = true;
-			$wp_taxonomies['category']->graphql_single_name = 'category';
-			$wp_taxonomies['category']->graphql_plural_name = 'categories';
+		if ( 'category' === $taxonomy ) {
+			$args['show_in_graphql']     = true;
+			$args['graphql_single_name'] = 'category';
+			$args['graphql_plural_name'] = 'categories';
+		} elseif ( 'post_tag' === $taxonomy ) { // Adds GraphQL support for tags.
+			$args['show_in_graphql']     = true;
+			$args['graphql_single_name'] = 'tag';
+			$args['graphql_plural_name'] = 'tags';
+		} elseif ( 'post_format' === $taxonomy ) { // Adds GraphQL support for post formats.
+			$args['show_in_graphql']     = true;
+			$args['graphql_single_name'] = 'postFormat';
+			$args['graphql_plural_name'] = 'postFormats';
 		}
 
-		// Adds GraphQL support for tags.
-		if ( isset( $wp_taxonomies['post_tag'] ) ) {
-			$wp_taxonomies['post_tag']->show_in_graphql     = true;
-			$wp_taxonomies['post_tag']->graphql_single_name = 'tag';
-			$wp_taxonomies['post_tag']->graphql_plural_name = 'tags';
+		return $args;
+	}
+
+	/**
+	 * This sets the post type /taxonomy GraphQL properties.
+	 *
+	 * @param array  $args      Array of arguments for registering the type.
+	 * @param string $wp_type   Post type / Taxonomy type key.
+	 *
+	 * @return array
+	 */
+	public static function set_wp_type_args( $args, $wp_type ) {
+
+		// Bail early if the post type is hidden from the WPGraphQL schema.
+		if ( empty( $args['show_in_graphql'] ) ) {
+			return $args;
 		}
 
-		// Adds GraphQL support for post formats.
-		if ( isset( $wp_taxonomies['post_format'] ) ) {
-			$wp_taxonomies['post_format']->show_in_graphql     = true;
-			$wp_taxonomies['post_format']->graphql_single_name = 'postFormat';
-			$wp_taxonomies['post_format']->graphql_plural_name = 'postFormats';
+		$defaults = [
+			// The "kind" of GraphQL type to register. Can be `interface`, `object`, or `union`.
+			'graphql_kind'                     => 'object',
+			// The callback used to resolve the type. Only used if `graphql_kind` is an `interface` or `union`.
+			'graphql_resolve_type'             => null,
+			// An array of custom interfaces the type should implement.
+			'graphql_interfaces'               => [],
+			// An array of default interfaces the type should exclude.
+			'graphql_exclude_interfaces'       => [],
+			// An array of custom connections the type should implement.
+			'graphql_connections'              => [],
+			// An array of default connection field names the type should exclude.
+			'graphql_exclude_connections'      => [],
```

Comment:
@jasonbahl should this be renamed to `graphql_exclude_connection_field_names`, to prevent users from thinking they can pass the "to" GraphQL type name (e.g. 'Comment' instead of 'comments'), or is this obvious?